### ISSUE TITLE
eosiocpp trough docker, file must be specified

### ISSUE THREAD

**Issue by paul-mesnilgrente** [EOSIO/eos#4397]
Hi everybody !

I have installed eosio through Docker. As said in the documentation, I've put these aliases:

```bash
alias runeos='docker run --rm \
--name eosio -d \
-p 8888:8888 \
-p 9876:9876 \
-v /tmp/work:/work \
-v /tmp/eosio/data:/mnt/dev/data \
-v /tmp/eosio/config:/mnt/dev/config \
eosio/eos-dev  /bin/bash -c \
"nodeos -e -p eosio \
--plugin eosio::wallet_api_plugin \
--plugin eosio::wallet_plugin \
--plugin eosio::producer_plugin \
--plugin eosio::history_plugin \
--plugin eosio::chain_api_plugin \
--plugin eosio::history_api_plugin \
--plugin eosio::http_plugin \
-d /mnt/dev/data \
--config-dir /mnt/dev/config \
--http-server-address=0.0.0.0:8888 \
--access-control-allow-origin=* \
--contracts-console"'
alias cleos='docker exec eosio /opt/eosio/bin/cleos --wallet-url http://localhost:8888'
```

And now, I am following the tutorial "tic tac toe" for the smart contracts. My problem is here: I have to compile `tic_tac_toe.hpp` into a `.abi` with this command: `eosiocpp -g tic_tac_toe.abi tic_tac_toe.hpp`. So I have set this alias: `alias eosiocpp='docker exec eosio /opt/eosio/bin/eosiocpp'` to have `eosiocpp` as a command. But the resulting output is only: `You must specify a file`.

I saw that the error is raised [here](https://github.com/EOSIO/eos/blob/master/tools/eosiocpp.in) on line 73. So I have checked my filename `tic_tac_toe.hpp` and it's alright. Is it because I am launching it through docker? So the third parameter is actually `/opt/eosio/bin/eosiocpp` in the script?

I would be happy to debug this problem but I am not very familiar with Docker.

**Comment by taokayan** [comment]
Can you try to compile a cpp file instead of a header file?

**Comment by paul-mesnilgrente** [comment]
Same result... I have tried directly in `eos/contracts/tic_tac_toe` instead of my personal files.

**Comment by paul-mesnilgrente** [comment]
It's impossible that I am the only one that is trying to compile its code using the docker image. Is there any documentation about this?

My configuration is:
**OS:** Ubuntu 18.04
**Docker:** Docker version 18.03.1-ce, build 9ee9f40

**Comment by liufuhu** [comment]
same question, can anyone tell us how to use ```eosiocpp```  command in docker?

**Comment by vijaytabhatt** [comment]
@paul-mesnilgrente  did you got any solution?

**Comment by paul-mesnilgrente** [comment]
No, I've managed to compile eos with 3.87GB of RAM, that's all I can say ^^

**Comment by kelonye** [comment]
Use `eosio/eos-dev:v1.1.1` for now.

**Comment by Justin-ye** [comment]
same error
